mobile/src/components/interview/components/history/history.tsx

```diff
@@ -1,9 +1,145 @@
 import React, { FC } from 'react';
 
-import { Text } from '~/components/common/common';
+import { ButtonVariant, DataStatus, PermissionKey } from '~/common/enums/enums';
+import { InterviewNoteCreateRequestDto } from '~/common/types/types';
+import {
+  Button,
+  Input,
+  ScrollView,
+  Stack,
+  Text,
+  View,
+} from '~/components/common/common';
+import { checkHasPermission } from '~/helpers/helpers';
+import {
+  useAppDispatch,
+  useAppForm,
+  useAppSelector,
+  useCallback,
+  useEffect,
+  useFocusEffect,
+  useState,
+} from '~/hooks/hooks';
+import { interviewActions } from '~/store/actions';
+import { interviewNotesCreateArguments } from '~/validation-schemas/validation-schemas';
+
+import { DEFAULT_CREATE_NOTE_PAYLOAD } from './common';
+import { NoteCardsList } from './components/components';
+import { styles } from './styles';
 
 const History: FC = () => {
-  return <Text>History</Text>;
+  const [isFormOpen, setIsFormOpen] = useState(false);
+  const dispatch = useAppDispatch();
+
+  const { user, interviewId, interviewDataStatus, notes } = useAppSelector(
+    ({ auth, interview }) => ({
+      user: auth.user,
+      notes: interview.notes,
+      interviewId: interview.interview?.id,
+      interviewDataStatus: interview.interviewDataStatus,
+    }),
+  );
+
+  const { control, errors, handleSubmit, reset } =
+    useAppForm<InterviewNoteCreateRequestDto>({
+      defaultValues: DEFAULT_CREATE_NOTE_PAYLOAD,
+      validationSchema: interviewNotesCreateArguments,
+    });
+
+  const isLoading = interviewDataStatus === DataStatus.PENDING;
+
+  const toggleInput = (): void => {
+    setIsFormOpen((prev) => !prev);
+    reset();
+  };
+
+  const hasPermission = checkHasPermission({
+    permissionKeys: [
+      PermissionKey.MANAGE_INTERVIEWS,
+      PermissionKey.MANAGE_INTERVIEW,
+    ],
+    userPermissions: user?.permissions ?? [],
+    checkMode: 'every',
+  });
```

Comment:
We don't need this here, user can't get to interview screen if he have no one of these permissions. FYI user is able to see this screen either with MANAGE_INTERVIEWS or MANAGE_INTERVIEW permission, the only difference is that user with MANAGE_INTERVIEW permission can see only interviews assigned to him
```suggestion
```

Reply:
The only thing that we need to check if user has MANAGE_INTERVIEW permission and interview is not assigned to him, then he shouldn't see add button

Reply:
And, when user has MANAGE_INTERVIEWS ...if I understand right, he should see add button on any interviews?

Reply:
Yes

Reply:
Sorry, Liza) But I'm confused.
If user has MANAGE_INTERVIEW,  he should see only interviews assigned to him ? Or should see all interviews...but notes can add only on interviews assigned to him ?

Reply:
He will see all interviews, but will be able to edit/change status/leave notes only if he is assigned to it. In interviews screen he will see only his interviews, it's handled by backend, so no changes on mobile side needed.